### ISSUE TITLE
The last element of math.range() does not always include the upper limit

### ISSUE THREAD

**Issue by arnog** [josdejong/mathjs#1429]
When passing the `includeEnd = true` flag to `math.range()`, the upper limit is not always included.

Example:
```
math.range(-3.2909135802469143, 3.2909135802469143, (3.2909135802469143 + 3.2909135802469143) / 10, true)
->
0: -3.2909135802469143
1: -2.6327308641975313
2: -1.9745481481481484
3: -1.3163654320987654
4: -0.6581827160493826
5: 2.220446049250313e-16
6: 0.658182716049383
7: 1.3163654320987659
8: 1.9745481481481488
9: 2.6327308641975318
```

but:
```
math.range(-3.2909135802469143, 3.2909135802469143,(3.2909135802469143 + 3.2909135802469143) / 9, true)
->
0: -3.2909135802469143
1: -2.5595994513031557
2: -1.8282853223593971
3: -1.0969711934156385
4: -0.36565706447187984
5: 0.36565706447187885
6: 1.0969711934156376
7: 1.8282853223593962
8: 2.559599451303155
9: 3.2909135802469134
```

**Comment by ericman314** [comment]
This looks like a problem due to round-off error. I think `range` was implemented with integers in mind. Perhaps it could be made more reliable for floating point values if it used a fuzzy comparison for the endpoint.

I think a better approach though would be to use what #1079 describes as a `linspace` function. _If_ such a function were implemented you could use it without worrying about round-off error:

```js
math.linspace(-3.2909135802469143, 3.2909135802469143, 10)
```

There's an open PR #1207 implementing this function, but it seems it's not quite finished yet. 

In the meantime, as an ugly hack you can add a very small amount to the `end` argument (such as half the step size), which will ensure that the end point is always included.

**Comment by josdejong** [comment]
An other way to solve this could be to use the `equal`, `smaller`, and `larger` functions inside `range`, these functions reckon with a configured `epsilon`.

@ericman314 this reminds me of an other (not related) open PR regarding round-off issues is #894 that you worked on. Would be great if you have some time to see if you can finish it?

**Comment by ericman314** [comment]
For sure, I can change the endpoint logic to use fuzzy comparisons.

On a separate note, is this behavior intentional?

```js
console.log(math.range(1, 5))           // [ 1, 2, 3, 4 ]
console.log(math.eval('range(1, 5)'))   // [ 1, 2, 3, 4, 5 ]
```

**Comment by josdejong** [comment]
Thanks for your PR's.

The difference between `math.range(1, 5) ` and `math.eval('range(1, 5)')` is indeed intentional. The JS API is zero base and has end of range excluded, the expression parser API is one based and has range ends included (behavior similar to Matlab). Here is a list with all the differences in behavior:

https://mathjs.org/docs/expressions/syntax.html#differences-from-javascript

**Comment by josdejong** [comment]
Just published `v5.7.0` containing the fix made by @ericman314, will close this issue now.
